.github/workflows/update-libs.yaml

```diff
@@ -0,0 +1,50 @@
+name: Auto-update Charm Libraries
+on:
+  workflow_call:
+    secrets:
+      CHARMHUB_TOKEN:
+        required: false
+      OBSERVABILITY_NOCTUA_TOKEN:
+        required: true
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-lib:
+    name: Check libraries
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
```

Comment:
Replacing the tag with a hash is something that the [renovate[bot] does](https://github.com/jnsgruk/zinc-k8s-operator/commit/2b6e50bbc9f1de4f423b9407148e56ecfd6e6811).
Without the bot I think we would be better off with human readable tags.